static/src/javascripts/projects/common/modules/atoms/youtube.js

```diff
@@ -375,6 +381,22 @@ const onPlayerReady = (
     }
 };
 
+const isAnyPlayerPlaying = (): boolean =>
+    Object.keys(players)
+        .map(key => players[key])
+        .filter((p: AtomPlayer): boolean => p.playing).length > 0;
+
+const triggerVideoStateEvent = (isPlaying: boolean): void => {
+    if (isPlaying) {
+        const videoPlaying = new Event('videoPlaying');
+        document.dispatchEvent(videoPlaying);
+    } else {
+        // Use videoEnded until tracker-js updated to videoStopped.
+        const videoStopped = new Event('videoEnded');
+        document.dispatchEvent(videoStopped);
+    }
```

Comment:
Ophan's `tracker-js` has code [that listens to these events](https://github.com/guardian/ophan/blob/59d4a6173512bcc2b481cb0b81d971dee92ff257/tracker-js/assets/coffee/ophan/attention.coffee#L105-L116), added in December 2014 with https://github.com/guardian/ophan/pull/717 (the corresponding PR in DotCom that generated these events being https://github.com/guardian/frontend/pull/7541).

The Guardian's widespread use of Youtube-hosted videos seems to have come around September 2016 with https://github.com/guardian/frontend/pull/14450 - then both https://github.com/guardian/frontend/pull/14560 in October and https://github.com/guardian/frontend/pull/14965 in November 2016 added video-completion tracking, so Ophan knew _how far into the video_ the user had got, but those changes didn't hook that into the attention-tracking code - so it's been broke for just over 3 years!